src/ui/molecules/stripe-express-checkout-element.svelte

```diff
@@ -22,36 +22,51 @@
   import type {
     ClickResolveDetails,
     StripeExpressCheckoutElementClickEvent,
+    StripeExpressCheckoutElementShippingAddressChangeEvent,
   } from "@stripe/stripe-js/dist/stripe-js/elements/express-checkout";
 
   export interface Props {
     onError: (error: StripeServiceError) => void | Promise<void>;
-    onReady: () => void | Promise<void>;
+    onReady?:
+      | (() => void | Promise<void>)
+      | ((
+          event: StripeExpressCheckoutElementReadyEvent,
+        ) => void | Promise<void>);
     onSubmit: (
       paymentMethod: string,
       emailValue: string,
+      event: StripeExpressCheckoutElementConfirmEvent,
     ) => void | Promise<void>;
+    onClick?: (event: StripeExpressCheckoutElementClickEvent) => void;
+    onShippingAddressChange?: (
+      event: StripeExpressCheckoutElementShippingAddressChangeEvent,
+    ) => void;
     elements: StripeElements;
     billingAddressRequired: boolean;
     forceEnableWalletMethods: boolean;
     expressCheckoutOptions?: StripeExpressCheckoutConfiguration;
+    hideOtherOptions?: boolean;
```

Comment:
It's not very clear that this will hide the other options separator, could we rename to:
```suggestion
    hideCheckoutSeparator?: boolean;
```

Another option is extracting the separator form this molecule